### PR TITLE
fix:  bug in variables ordering in NormalizedReluBounding

### PR DESCRIPTION
**Pull request by lzampier**
A bug affects `anemoi.models.layers.bounding.NormalizedReluBounding` if the order of the variable douring the call in config does not match the order of the variables in the torch array. This a problem emerged only when multiple variables were bounded at the same time with this procedure.

This is fixed by making sure that `self.norm_min_val` follows the same order as `self.data_index`.

### CODE REVIEW

File: models/src/anemoi/models/layers/bounding.py [hunk 1]
```diff
@@ -153,6 +155,9 @@ def __init__(
                 std = self.statistics["stdev"][stat_index]
                 self.norm_min_val[ii] = min_val[ii] / std
 
+        # Reorder normalized min values based on data_index
+        self.norm_min_val = norm_min_val[torch.argsort(torch.tensor(self.data_index))]
```
Comment (sahahner) [inline models/src/anemoi/models/layers/bounding.py]: `self.data_index` should lie on the CPU, so I would use numpy operations here.

Comment (lzampier) [reply models/src/anemoi/models/layers/bounding.py]: Good point! Now using `np.array` and `np.argsort`.

**Comment by lzampier** [comment]
The effect of the bug does only show in the training metrics. I can write a test that fails if the order of variables is wrong after the initialization.

**Comment by lzampier** [comment]
> Would it make sense to add a test that would catch this bug next time?

Yes, more tests would be good. I would postpone this to a different pull request, given that we will refactor the interface of this function to make it simpler over the next days. The tests will also inevitably change for the refactor.

**Review by sahahner — APPROVED** [review]
Looks good.
